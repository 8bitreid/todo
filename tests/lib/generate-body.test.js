const generateBody = require('../../lib/generate-body')
const payloads = require('../fixtures/payloads')

describe('generate-body', () => {
  let context

  const title = 'Jason!'
  const file = 'index.js'
  const contents = '\n\n@todo Jason!\nasdfas\nasdfdsafasd\nsd\nasdfsa\n\nsdfsadfsa'
  const contentsBody = '\n\n@todo Jason!\n@body This one has a body\nasdfas\nasdfdsafasd\nsd\nasdfsa\n\nsdfsadfsa'
  const author = payloads.basic.payload.pusher.name
  const sha = payloads.basic.payload.head_commit.id

  beforeEach(() => {
    context = {
      repo: () => ({
        owner: payloads.basic.payload.repository.owner.login,
        repo: payloads.basic.payload.repository.name
      }),
      payload: {
        installation: {
          id: 10000
        }
      }
    }
  })

  it('generates a body string', () => {
    const config = { keyword: '@todo', bodyKeyword: '@body', blobLines: 5 }
    const body = generateBody(context, config, title, file, contents, author, sha)
    expect(body).toMatchSnapshot()
  })

  describe('autoAssign', () => {
    it('prepares a message with the committer', () => {
      const config = { keyword: '@todo', bodyKeyword: '@body', autoAssign: true, blobLines: 5 }
      const body = generateBody(context, config, title, file, contents, author, sha)
      expect(body).toMatchSnapshot()
    })

    it('prepares a message without an assignee', () => {
      const config = { keyword: '@todo', bodyKeyword: '@body', autoAssign: false, blobLines: 5 }
      const body = generateBody(context, config, title, file, contents, author, sha)
      expect(body).toMatchSnapshot()
    })

    it('prepares a message with the configured assignee', () => {
      const config = { keyword: '@todo', bodyKeyword: '@body', autoAssign: '@matchai', blobLines: 5 }
      const body = generateBody(context, config, title, file, contents, author, sha)
      expect(body).toMatchSnapshot()
    })

    it('prepares a message with the configured assignees', () => {
      const config = { keyword: '@todo', bodyKeyword: '@body', autoAssign: ['@JasonEtco', 'matchai', 'defunkt'], blobLines: 5 }
      const body = generateBody(context, config, title, file, contents, author, sha)
      expect(body).toMatchSnapshot()
    })
  })

  it('generates a body string with a PR', () => {
    const config = { keyword: '@todo', bodyKeyword: '@body', blobLines: 5 }
    const body = generateBody(context, config, title, file, contents, author, sha, 10)
    expect(body).toMatchSnapshot()
  })

  describe('blobLines', () => {
    it('generates a body string without a blob, blobLines: false', () => {
      const config = { keyword: '@todo', bodyKeyword: '@body', blobLines: false }
      const body = generateBody(context, config, title, file, contents, author, sha)
      expect(body).toMatchSnapshot()
    })

    it('generates a body string without a blob, blobLines: 0', () => {
      const config = { keyword: '@todo', bodyKeyword: '@body', blobLines: 0 }
      const body = generateBody(context, config, title, file, contents, author, sha)
      expect(body).toMatchSnapshot()
    })
  })

  it('generates a body string with a custom message', () => {
    const config = { keyword: '@todo', bodyKeyword: '@body', blobLines: 5 }
    const body = generateBody(context, config, title, file, contentsBody, author, sha)
    expect(body).toMatchSnapshot()
  })

  it('generates a body string with a custom message but no blob', () => {
    const config = { keyword: '@todo', bodyKeyword: '@body', blobLines: false }
    const body = generateBody(context, config, title, file, contentsBody, author, sha)
    expect(body).toMatchSnapshot()
  })

  it('respects the configured bodyKeyword', () => {
    const config = { keyword: '@todo', bodyKeyword: '@pizza', blobLines: 5 }
    const contentsPizza = '\n\n@todo Jason!\n@pizza This one has a body\n@body eat some pizza\nasdfas\nasdfdsafasd\nsd\nasdfsa\n\nsdfsadfsa'
    const body = generateBody(context, config, title, file, contentsPizza, author, sha)
    expect(body).toMatchSnapshot()
  })
})
