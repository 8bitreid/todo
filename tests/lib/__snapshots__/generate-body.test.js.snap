// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-body autoAssign prepares a message with the committer 1`] = `
"https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body autoAssign prepares a message with the configured assignee 1`] = `
"https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been automagically assigned to @matchai.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body autoAssign prepares a message with the configured assignees 1`] = `
"https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been automagically assigned to @JasonEtco, @matchai and @defunkt.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body autoAssign prepares a message without an assignee 1`] = `
"https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body blobLines generates a body string without a blob, blobLines: 0 1`] = `
"###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body blobLines generates a body string without a blob, blobLines: false 1`] = `
"###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body generates a body string 1`] = `
"https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body generates a body string with a PR 1`] = `
"## Jason!

https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This comment was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2 in #10. cc @JasonEtco.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body generates a body string with a custom message 1`] = `
"This one has a body

---

https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body generates a body string with a custom message but no blob 1`] = `
"This one has a body

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;

exports[`generate-body respects the configured bodyKeyword 1`] = `
"This one has a body

---

https://github.com/JasonEtco/test/blob/f7d286aa6381bbb5045288496403d9427b0746e2/index.js#L3-L8

---

###### This issue was generated by [todo](https://todo.jasonet.co) based on a \`@todo\` comment in f7d286aa6381bbb5045288496403d9427b0746e2. It's been assigned to @JasonEtco because they committed the code.

<!-- probot = {\\"10000\\":{\\"title\\":\\"Jason!\\",\\"file\\":\\"index.js\\",\\"author\\":\\"JasonEtco\\",\\"sha\\":\\"f7d286aa6381bbb5045288496403d9427b0746e2\\"}} -->"
`;
